### PR TITLE
Autocomplete Suggestion Improvements

### PR DESCRIPTION
**Pull request by nlogozzo**
- Suggestions will be shown case-insensitivly 
- Use down arrow to navigation suggestions
- Use enter to select first suggestion if none selected, otherwise selected suggestion

### CODE REVIEW

File: NickvisionMoney.GNOME/Views/TransactionDialog.cs [hunk 1]
```diff
@@ -245,12 +246,36 @@ private TransactionDialog(Gtk.Builder builder, TransactionDialogController contr
...
+                }
+            }
+            if(e.Keyval == 65364) //down arrow
+            {
+                if(_autocompleteBox.GetVisible())
+                {
+                    _autocompleteBox.GrabFocus();
+                    return true;
+                }
+            }
```
Comment (DaPigGuy) [inline NickvisionMoney.GNOME/Views/TransactionDialog.cs:276]:
```suggestion
            if(_autocompleteBox.GetVisible())
            {
                if(e.Keyval == 65293 || e.Keyval == 65421) //enter | keypad enter
                {
                    _autocompleteBox.AcceptSuggestion(0);
                    return true;
                }
                if(e.Keyval == 65364) //down arrow
                {
                    _autocompleteBox.GrabFocus();
                    return true;
                }
            }
```

File: NickvisionMoney.GNOME/Views/TransactionDialog.cs [hunk 2]
```diff
@@ -243,14 +246,43 @@ private TransactionDialog(Gtk.Builder builder, TransactionDialogController contr
...
-            if(e.Flags.HasFlag(Gtk.StateFlags.FocusWithin) && !_descriptionRow.GetStateFlags().HasFlag(Gtk.StateFlags.FocusWithin))
+            if(e.Flags.HasFlag(Gtk.StateFlags.FocusWithin) && !_descriptionRow.GetStateFlags().HasFlag(Gtk.StateFlags.FocusWithin) && _canHideAutobox)
             {
+                _descriptionRow.SetActivatesDefault(true);
                 _autocompleteBox.SetVisible(false);
             }
+            if(!_canHideAutobox)
+            {
+                _canHideAutobox = true;
+            }
```
Comment (DaPigGuy) [inline NickvisionMoney.GNOME/Views/TransactionDialog.cs:257]:
```suggestion
            if(!_canHideAutobox)
            {
                _canHideAutobox = true;
            }
            else if(e.Flags.HasFlag(Gtk.StateFlags.FocusWithin) && !_descriptionRow.GetStateFlags().HasFlag(Gtk.StateFlags.FocusWithin))
            {
                _descriptionRow.SetActivatesDefault(true);
                _autocompleteBox.SetVisible(false);
            }
```

File: NickvisionMoney.GNOME/Views/TransactionDialog.cs (hunk 1 above)
Comment (nlogozzo) [reply NickvisionMoney.GNOME/Views/TransactionDialog.cs]: I don't want to do it like this incase we need to keys for something else and need to add more if statements. Best to keep the outside if statements just comparing keyvals as it is a OnKeyPressed handler.

**Comment by nlogozzo** [comment]
Ok, ready. See PR description for changes

**Comment by nlogozzo** [comment]
https://github.com/NickvisionApps/Denaro/pull/555/commits/1c9a9a81dc5d07de24ba24e3117e15b1b4650920

15 is odd-number, let's keep it even 😅

**Review by fsobolev — APPROVED** [review]

**Pull request merged**